### PR TITLE
Update miniconda from 4.7.12 to py37_4.8.2

### PR DESCRIPTION
**Pull request by b0ttle**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [ ] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

**Comment by b0ttle** [comment]
btw miniconda divided into version py37 and py38, i chose py37 here

**Comment by ran-dall** [comment]
@b0ttle This might be a stupid question, but what's the difference?

**Comment by b0ttle** [comment]
@ran-dall Sorry I didn't know the difference either, but in miniconda's download page [https://repo.anaconda.com/miniconda/](url), they are different files.

**Comment by ran-dall** [comment]
@b0ttle It probably has something to do with `py_37` being based on `3.7.x` branch of Python and py_38` being based on `3.8.x` branch of Python. However, I don't know which would be better for HBC, or if we should now have two different casks.

@vitorgalvao Could I get your opinion on this situation?

**Comment by b0ttle** [comment]
@ran-dall Actually there exists a file named 'latest', which is the same as 'py37'(Same MD5). Maybe we can change the download link to the 'latest' version. It also can let users to know when to send a update PR. In 3 month no one notices miniconda get updated.

**Comment by ran-dall** [comment]
@b0ttle We'd like to use a version where we can, so I think it's better to keep it. It makes sense the `py37` is the `latest` because it's probably the most compatible (and probably has fewer bugs). With that considered, I think this update makes sense; but let's see what some of the other maintainers think.

> In 3 month no one notices miniconda get updated.

I can address with the ML. So it's not going to happen as often.

**Comment by vitorgalvao** [comment]
This cask should be updated to use the `pkg` installer instead. It will allow us a better `uninstall`.

Is that something you want to pursue, @b0ttle?

**Comment by b0ttle** [comment]
@vitorgalvao I always use `.sh` for conda's install, not sure if it will be better using `pkg`. Depends on your judgement :).

**Comment by vitorgalvao** [comment]
> I always use `.sh` for conda's install, not sure if it will be better using `pkg`.

As a manual install, it’s a matter of personal preference. As a cask, the `pkg` is better.

**Comment by ran-dall** [comment]
@b0ttle If you don't want to do or you need some help, I'll be happy to help / do it. Just let me know...

**Comment by b0ttle** [comment]
@ran-dall It is difficult for me to communicate in English. Sometimes I can't get your ideas, sorry about that.
So help me do it, thank you very much.

**Comment by ran-dall** [comment]
@b0ttle Completely understandable. I'll try to keep it as "simple" English as possible. 😉

Let me look over what needs to get done, and I'll post it here shortly. All you'd need to do is copy the changes over and submit to this PR.

File: Casks/miniconda.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   # repo.anaconda.com/miniconda/ was verified as official when first introduced to the cask
-  url "https://repo.anaconda.com/miniconda/Miniconda3-#{version}-MacOSX-x86_64.sh"
+  url "https://repo.anaconda.com/miniconda/Miniconda3-py37_#{version}-MacOSX-x86_64.sh"
   name 'Continuum Analytics Miniconda'
   homepage 'https://conda.io/miniconda.html'
 
   auto_updates true
   container type: :naked
 
   installer script: {
```
Comment (ran-dall) [inline Casks/miniconda.rb:13]:
```ruby
  installer script: {
                      executable: "Miniconda3-py37_#{version}-MacOSX-x86_64.sh",
                      args:       ['-b', '-p', "#{caskroom_path}/base"],
                    }
  binary "#{caskroom_path}/base/condabin/conda"

  uninstall delete: "#{caskroom_path}/base"
```

Replace with:

```ruby
  pkg 'Miniconda3-latest-MacOSX-x86_64.pkg'

  uninstall pkgutil: [
                       'io.continuum.pkg.apreinstall',
                       'io.continuum.pkg.conda.exe',
                       'io.continuum.pkg.pathupdate',
                       'io.continuum.pkg.postextract',
                       'io.continuum.pkg.preconda',
                     ]

```

**Review by ran-dall — CHANGES_REQUESTED** [review]

File: Casks/miniconda.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-                      executable: "Miniconda3-#{version}-MacOSX-x86_64.sh",
+                      executable: "Miniconda3-py37_#{version}-MacOSX-x86_64.sh",
                       args:       ['-b', '-p', "#{caskroom_path}/base"],
                     }
   binary "#{caskroom_path}/base/condabin/conda"
```
Comment (ran-dall) [inline Casks/miniconda.rb:17]:
```suggestion
  pkg 'Miniconda3-latest-MacOSX-x86_64.pkg'
```

**Review by ran-dall — COMMENTED** [review]
It wouldn't let me a the suggestion on line 19, but replace line 19 with:

```
  uninstall pkgutil: [
                       'io.continuum.pkg.apreinstall',
                       'io.continuum.pkg.conda.exe',
                       'io.continuum.pkg.pathupdate',
                       'io.continuum.pkg.postextract',
                       'io.continuum.pkg.preconda',
                     ]
```

The `zap` stanza and the rest of it should be left the same.

File: Casks/miniconda.rb [hunk 3]
```diff
@@ -1,17 +1,17 @@
 cask 'miniconda' do
-  version 'py37_4.8.2'
+  version '4.8.2'
   sha256 'd1fca4f74f9971c27220122723843f6c879a5d13ff59c01fca17ef62a1576732'
 
   # repo.anaconda.com/miniconda/ was verified as official when first introduced to the cask
-  url "https://repo.anaconda.com/miniconda/Miniconda3-#{version}-MacOSX-x86_64.sh"
+  url "https://repo.anaconda.com/miniconda/Miniconda3-py37_#{version}-MacOSX-x86_64.sh"
```
[the remark this answers is not in the document]
Comment (vitorgalvao) [reply Casks/miniconda.rb:6]: You need to suggest the whole line, including `url` and the quotes. Is there no versioned link for the PKG?

**Comment by ran-dall** [comment]
@vitorgalvao Does this still require `container type: :naked` now that we're using a `pkg`?

**Comment by vitorgalvao** [comment]
> @vitorgalvao Does this still require `container type: :naked` now that we're using a `pkg`?

Nope.

File: Casks/miniconda.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (ran-dall) [reply Casks/miniconda.rb:6]: @vitorgalvao Thanks, I think this is my first time using suggestions. I like to try new things.

File: Casks/miniconda.rb [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
   name 'Continuum Analytics Miniconda'
   homepage 'https://conda.io/miniconda.html'
 
   auto_updates true
   container type: :naked
```
Comment (ran-dall) [inline Casks/miniconda.rb:11]:
```suggestion

```

Comment (ran-dall) [reply Casks/miniconda.rb:11]: > > @vitorgalvao Does this still require `container type: :naked` now that we're using a `pkg`?
> 
> Nope.

@b0ttle So you can also remove line 11 completely.

**Comment by ran-dall** [comment]
@b0ttle So the final should look something like this (you can just copy and paste as a whole, if you want)

```ruby
cask 'miniconda' do
  version '4.8.2'
  sha256 'f3ede3a58d82fb5dcbca52d291a9edb5cd962d84d823a20693dd4bb27506cdd0'

  # repo.anaconda.com/miniconda/ was verified as official when first introduced to the cask
  url "https://repo.anaconda.com/miniconda/Miniconda3-#{version}-MacOSX-x86_64.pkg"
  name 'Continuum Analytics Miniconda'
  homepage 'https://conda.io/miniconda.html'

  auto_updates true

  pkg 'Miniconda3-latest-MacOSX-x86_64.pkg'

  uninstall pkgutil: [
                       'io.continuum.pkg.apreinstall',
                       'io.continuum.pkg.conda.exe',
                       'io.continuum.pkg.pathupdate',
                       'io.continuum.pkg.postextract',
                       'io.continuum.pkg.preconda',
                     ]

  zap trash: [
               '~/.condarc',
               '~/.conda',
               '~/.continuum',
             ]

  caveats <<~EOS
    Please run the following to setup your shell:

      conda init "$(basename "${SHELL}")"
  EOS
end
```

**Comment by b0ttle** [comment]
@ran-dall Already add your code in newest commit! Thank you for your kindly help and patients. Will try to contribute more in the future!

**Comment by ran-dall** [comment]
@b0ttle Thank you for your effort and work. We'd be very happy to have another frequent contributor.

Also, I don't think the caveat is required anymore with the pkg.

**Review by ran-dall — APPROVED** [review]

**Pull request merged**